### PR TITLE
Improve organization of the DataAnnotation related stuff

### PR DESCRIPTION
**Pull request by zvirja**
In this commit I perform a bit of cleanup to make the code structure more clear and consistent:
- all the Xeger stuff was moved to its own namespace;
- all the data annotations specific stuff has been moved under the `DataAnnotations` namespace. While some of the changes are broken, we are fine as those files were introduced in non-released v4;
- moved `NumericRangedRequestRelay` to Customization section for consistency with other builders. Now all the builders are located nearby;
- extended the `NoDataAnnotationsCustomization` to remove all the related builders.

All those changes are minor, but now it's more clear which particular stuff is specific to DataAnnotations.

### CODE REVIEW

**Comment by zvirja** [comment]
@moodmosaic The issue #938 is about Xeger only, while in this PR I perform other improvements as well. Those rest changes don't depend on our decision and still should be applied.

I'd vote to proceed with this PR. Even if later we decide to remove Xeger from our sources, I believe it will not happen in the very nearby future, as we should perform a lot of change there (port to .NET Standard, etc).

File: Src/AutoFixture/Xeger/Automaton.cs [hunk 1]
```diff
@@ -34,7 +34,7 @@
 using System.Collections.Generic;
 using System.Linq;
 
-namespace AutoFixture.DataAnnotations
+namespace AutoFixture.Xeger
```
Comment (moodmosaic) [inline Src/AutoFixture/Xeger/Automaton.cs]: There's no such thing as `AutoFixture.Xeger`. I think we should either revert this change, or change it to `namespace Fare`, until we're done with #938.

**Comment by zvirja** [comment]
Fixed - moved the Xeger related stuff under the `Fare` namespace 😉

P.S. Sorry for the late follow-up - was (and still am a bit) sick.

File: Src/AutoFixture/Fare/Xeger.cs [hunk 2]
```diff
@@ -18,15 +18,18 @@
...
-namespace AutoFixture.DataAnnotations
+namespace Fare
 {
     /// <summary>
     /// An object that will generate text from a regular expression. In a way, 
     /// it's the opposite of a regular expression matcher: an instance of this class
     /// will produce text that is guaranteed to match the regular expression passed in.
     /// </summary>
+    [SuppressMessage("Microsoft.Naming", "CA1724:TypeNamesShouldNotMatchNamespaces", 
+        Justification = "This namespace contains everything specific to this module implementation, so call it the same.")]
```
Comment (moodmosaic) [inline Src/AutoFixture/Fare/Xeger.cs]: Note sure why this suppression is needed.

File: Src/AutoFixture/Fixture.cs [hunk 3]
```diff
@@ -119,14 +119,17 @@ public Fixture(ISpecimenBuilder engine, MultipleRelay multiple)
...
-                                new StringLengthAttributeRelay(),
-                                new RegularExpressionAttributeRelay(),
                                 new EnumGenerator(),
                                 new LambdaExpressionGenerator(),
                                 CreateDefaultValueBuilder(CultureInfo.InvariantCulture),
                                 CreateDefaultValueBuilder(Encoding.UTF8),
-                                CreateDefaultValueBuilder(IPAddress.Loopback))),
+                                CreateDefaultValueBuilder(IPAddress.Loopback),
+                        
+                                /* Data annotations */
```
Comment (moodmosaic) [inline Src/AutoFixture/Fixture.cs]: There must be a better way. :)

**Comment by moodmosaic** [comment]
>P.S. Sorry for the late follow-up - was (and still am a bit) sick.

No worries! No rush at all. (Get well soon!) 👍 :1st_place_medal: 💯 🚀

File: Src/AutoFixture/Fare/Xeger.cs (hunk 2 above)
Comment (zvirja) [reply Src/AutoFixture/Fare/Xeger.cs]: Nice catch 🏇 That was a left-over from the previous implementation and now it's indeed not required more. Removed it.

File: Src/AutoFixture/Fixture.cs (hunk 3 above)
Comment (zvirja) [reply Src/AutoFixture/Fixture.cs]: Yep. That is a temporary approach (probably). Later when we have a few more implementations, we could see whether it makes sense to organize them to a kind of standalone builder or so. For now it isn't clear how to deal with that to not loose the flexibility of potential customization... Any way, that change is out of scope of this PR.

Comment (moodmosaic) [reply Src/AutoFixture/Fixture.cs]: I've opened https://github.com/AutoFixture/AutoFixture/issues/944 so we don't easily forget it.
